owners/index.js

```diff
@@ -34,85 +36,55 @@ module.exports = app => {
     level: process.env.LOG_LEVEL || 'info',
   });
 
-  /**
-   * Runs the steps to create or update an owners-bot check-run on a GitHub Pull
-   * Request.
-   *
-   * @param {!GitHub} github GitHub API interface.
-   * @param {!PullRequest} pr pull request to run owners check on.
-   */
-  async function runOwnersCheck(github, pr) {
-    const localRepo = new LocalRepository(process.env.GITHUB_REPO_DIR);
-    const ownersCheck = new OwnersCheck(localRepo, github, pr);
-    let checkRunId;
-    let latestCheckRun;
+  /** Health check server endpoints **/
+  // TODO(rcebulko): Implement GitHub authentication to prevent spamming any of
+  // these endpoints.
+  router.get('/status', (req, res) => {
+    res.send('The OWNERS bot is live and running!');
+  });
 
-    try {
-      checkRunId = await github.getCheckRunId(pr.headSha);
-      latestCheckRun = await ownersCheck.run();
-    } catch (error) {
-      // If anything goes wrong, report a failing check.
-      latestCheckRun = new CheckRun(
-        'The check encountered an error!',
-        'OWNERS check encountered an error:\n' + error
-      );
-    }
+  router.get('/tree', async (req, res) => {
+    const parser = new OwnersParser(localRepo, req.log);
+    const ownersTree = await parser.parseOwnersTree();
 
-    if (checkRunId) {
-      await github.updateCheckRun(checkRunId, latestCheckRun);
-    } else {
-      // We need to add a delay on the PR creation and check creation since
-      // GitHub might not be ready.
-      // TODO: Verify this is still needed.
-      await sleep(GITHUB_CHECKRUN_DELAY);
-      await github.createCheckRun(pr.headSha, latestCheckRun);
-    }
-  }
+    res.send(`<pre>${ownersTree.toString()}</pre>`);
+  });
 
-  /**
-   * Runs the steps to create or update an owners-bot check-run on a GitHub Pull
-   * Request.
-   *
-   * @param {!GitHub} github GitHub API interface.
-   * @param {!number} prNumber pull request number.
-   */
-  async function runOwnersCheckOnPrNumber(github, prNumber) {
-    const pr = await github.getPullRequest(prNumber);
-    await runOwnersCheck(github, pr);
-  }
+  router.get('/check/:prNumber', async (req, res) => {
+    const octokit = new Octokit({auth: `token ${GITHUB_TOKEN}`});
+    const github = new GitHub(octokit, 'ampproject', 'amphtml', app.log);
```

Comment:
Flags instead of hard coded repo slug? This way you can test it on a fork when you want to make serious changes

Reply:
Done.